cheddar/cheddar-metrics-intercom/src/test/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollectorTest.java

```diff
@@ -0,0 +1,172 @@
+public class IntercomMetricCollectorTest {
+
+    private IntercomMetricCollector intercomMetricCollector;
+    private List<Class<?>> classTypes;
+    private List<CustomAttribute<?>> expectedAttributes;
+
+    @Before
+    public void setUp() {
+        PowerMockito.mockStatic(CustomAttribute.class);
+        PowerMockito.mockStatic(User.class);
+        classTypes = Arrays.asList(Boolean.class, String.class, Integer.class, Double.class, Long.class, Float.class);
+        expectedAttributes = new ArrayList<>();
+        final String personalAccessToken = randomString();
+        intercomMetricCollector = new IntercomMetricCollector(personalAccessToken);
+    }
+
+    @Test
+    public void shouldAddCustomAttributesToUser_withUserIdAndValidAttribute() {
+        // Given
+        final String userId = randomId();
+        final User mockUser = mock(User.class);
+        when(User.find(userId)).thenReturn(mockUser);
+        final Map<String, Object> customAttributes = mockCustomAttributes();
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        final ArgumentCaptor<CustomAttribute> argumentCaptor = ArgumentCaptor.forClass(CustomAttribute.class);
+        verify(mockUser, times(classTypes.size())).addCustomAttribute(argumentCaptor.capture());
+        assertThat(argumentCaptor.getAllValues(),
+                containsInAnyOrder(expectedAttributes.toArray(new CustomAttribute[expectedAttributes.size()])));
+        verifyStatic();
+        User.update(mockUser);
+    }
+
+    @Test
+    public void shouldAddValidCustomAttributesToUser_withValidAndInvalidAttribute() {
+        // Given
+        final String userId = randomId();
+        final User mockUser = mock(User.class);
+        when(User.find(userId)).thenReturn(mockUser);
+        final Map<String, Object> customAttributes = mockCustomAttributes();
+        customAttributes.put(randomString(), new Object());
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        final ArgumentCaptor<CustomAttribute> argumentCaptor = ArgumentCaptor.forClass(CustomAttribute.class);
+        verify(mockUser, times(classTypes.size())).addCustomAttribute(argumentCaptor.capture());
+        assertThat(argumentCaptor.getAllValues(),
+                containsInAnyOrder(expectedAttributes.toArray(new CustomAttribute[expectedAttributes.size()])));
+        verifyStatic();
+        User.update(mockUser);
+    }
+
+    @Test
+    public void shouldNotAddCustomAttributesToUser_withIntercomFindUserException() {
+        // Given
+        final String userId = randomId();
+        when(User.find(userId)).thenThrow(Exception.class);
+        final Map<String, Object> customAttributes = mockCustomAttributes();
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        verifyStatic(never());
```

Comment:
I can't see what in the code prevents the `User.update` method from being called.